Codesign macOS builds

vadi2 opened the pull request:
<!-- Keep the title short & concise so your mom can understand it -->
#### Brief overview of PR changes/additions
Codesign macOS builds in Github Actions.
#### Motivation for adding to Mudlet
So macOS builds have a (bit) less warnings and importantly, can self-update.
#### Other info (issues closed, discussion etc)
Fix https://github.com/Mudlet/Mudlet/issues/4745

Complimentary PR: https://github.com/Mudlet/installers/pull/98

Ditched Cocoapods since latest Sparkle with the codesigning fix for fr_CA and pt translations is not available there. It also removes a package manager we don't know from our mix!

vadi2 committed:
Update build-mudlet.yml

vadi2 committed:
Update travis.osx.after_success.sh

vadi2 committed:
Consolidate packaging steps

vadi2 commented:
Bizarre error from Apple tooling:

> ./Mudlet-4.10.1-testing-pr4746-b27fb709.app: main executable failed strict validation
In subcomponent: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-b27fb709.app/Contents/Frameworks/QtCore.framework

SlySven commented:
Have we not donated enough flesh/blood/gold to the Apple God? 🤣

vadi2 commented:
That is a good point - contacted their support about this.

I wonder if it's because we are using official Qt release builds now, which themselves might be signed already... ?

vadi2 committed:
Use working version of aqtinstall

vadi2 committed:
Purge Qt cache

vadi2 committed:
Merge remote-tracking branch 'origin/development' into codesign-macos-builds

vadi2 committed:
Merge remote-tracking branch 'origin/development' into codesign-macos-builds

vadi2 committed:
Merge remote-tracking branch 'origin/codesign-macos-builds' into codesign-macos-builds

vadi2 commented:
Current outstanding issue - French-Canadian and Portugese translations are somehow being added post-sign step:

```
% codesign -vv --deep-verify Mudlet-4.10.1-testing-pr4746-2592d773.app 
Mudlet-4.10.1-testing-pr4746-2592d773.app: a sealed resource is missing or invalid
In subcomponent: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/Sparkle.strings
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUAutomaticUpdateAlert.nib
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUUpdateAlert.nib
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUUpdatePermissionPrompt.nib
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/Sparkle.strings
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUAutomaticUpdateAlert.nib
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUUpdatePermissionPrompt.nib
file added: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUUpdateAlert.nib
file missing: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj
file missing: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-2592d773.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj
```

vadi2 committed:
Minor script tweaks

vadi2 committed:
Checkout the debug version of code signing

SlySven commented:
> Current outstanding issue - French-Canadian and Portugese translations are somehow being added post-sign step:

Those are not locales we explicit include in our own code (IIRC we have `fr_FR` French(France) and `pt_PT` Portuguese(Portugal)) - are there any other locales in the Sparkle framework that we do ***not*** use for Mudlet? If not then I guess we have a build system issue and we might want to try culling them from there...

SlySven commented on CI/travis.osx.after_success.sh:
```diff
@@ -21,27 +21,21 @@ if [ "${DEPLOY}" = "deploy" ]; then
-  # as Travis does not allow signing on usual PR builds
-  if [ ! -z "$MACOS_SIGNING_PASS" ]; then
+  # as CI's don't allow signing on usual PR builds
+  if [ -n "$MACOS_SIGNING_PASS" ]; then
```
ShellCheck pick that up? :smiling_imp:

vadi2 replied:
Yeah...

SlySven commented:
I note that you are frobbing `./CI/travis.osx.after_success.sh` - but that *seems* to be a Travis CI file and I thought we were not using (could not use) Travis anymore since they have pulled (the rug) out of (from under) supporting FOSS projects? :confused:

vadi2 commented:
Yes, we don't use Travis anymore - the file just kept the name not to break things until we get it ironed out.

vadi2 commented:
Removing those two translations didn't help:

```
% codesign -vv --deep-verify Mudlet-4.10.1-testing-pr4746-b17ae072.app 
Mudlet-4.10.1-testing-pr4746-b17ae072.app: a sealed resource is missing or invalid
In subcomponent: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-b17ae072.app/Contents/Frameworks/Sparkle.framework
file missing: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-b17ae072.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj
file missing: /Users/Vadim/Desktop/Mudlet-4.10.1-testing-pr4746-b17ae072.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj
```

vadi2 committed:
Try with 1.23

vadi2 committed:
Back to Sparkle 1.18 which is what worked in 4.10

vadi2 committed:
Actually install 1.18

vadi2 committed:
Try 1.23

1.18 gives:

 ./Mudlet-4.10.1-testing-pr4746-2dc0ce54.app: bundle format is ambiguous (could be app or framework)
In subcomponent: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-2dc0ce54.app/Contents/Frameworks/Sparkle.framework

vadi2 committed:
Try 1.22

1.23 gave: Verifying codesigning...
./Mudlet-4.10.1-testing-pr4746-5f005b07.app: a sealed resource is missing or invalid
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/Sparkle.strings
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUAutomaticUpdateAlert.nib
In subcomponent: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUUpdatePermissionPrompt.nib
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj/SUUpdateAlert.nib
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/Sparkle.strings
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUAutomaticUpdateAlert.nib
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUUpdatePermissionPrompt.nib
file added: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj/SUUpdateAlert.nib
file missing: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj
file missing: /Users/runner/work/Mudlet/b/ninja/Mudlet-4.10.1-testing-pr4746-5f005b07.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj

vadi2 committed:
Try 1.21

vadi2 committed:
Back up to 1.24

vadi2 committed:
Enable tmate debugging

vadi2 committed:
Pass all necessary environment variables to tmate

vadi2 committed:
Don't delete the identity

vadi2 committed:
Fix up the timeout

vadi2 commented:
Codesigning the test app works:

```
Mac-1613721063049:Sparkle-1.24.0.tar runner$ ls
CHANGELOG                       SampleAppcast.xml               Sparkle Test App.app.dSYM       Sparkle.framework.dSYM
LICENSE                         Sparkle Test App.app            Sparkle.framework               bin
Mac-1613721063049:Sparkle-1.24.0.tar runner$ ls Sparkle\ Test\ App.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj
SUAutomaticUpdateAlert.nib      SUUpdateAlert.nib               SUUpdatePermissionPrompt.nib    Sparkle.strings
Mac-1613721063049:Sparkle-1.24.0.tar runner$ rm -f Sparkle\ Test\ App.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/pt.lproj
Mac-1613721063049:Sparkle-1.24.0.tar runner$ rm -f Sparkle\ Test\ App.app/Contents/Frameworks/Sparkle.framework/Versions/Current/Resources/fr_CA.lproj
Mac-1613721063049:Sparkle-1.24.0.tar runner$ codesign -f -s "-" Sparkle\ Test\ App.app/Contents/Frameworks/Sparkle.framework/Versions/A/Sparkle
Sparkle Test App.app/Contents/Frameworks/Sparkle.framework/Versions/A/Sparkle: replacing existing signature
Mac-1613721063049:Sparkle-1.24.0.tar runner$ codesign -f -s "-" Sparkle\ Test\ App.app
Sparkle Test App.app: replacing existing signature
Mac-1613721063049:Sparkle-1.24.0.tar runner$ codesign --verify --deep -vvv Sparkle\ Test\ App.app
--prepared:/Users/runner/Sparkle/Sparkle-1.24.0.tar/Sparkle Test App.app/Contents/Frameworks/Sparkle.framework/Versions/Current/.
--validated:/Users/runner/Sparkle/Sparkle-1.24.0.tar/Sparkle Test App.app/Contents/Frameworks/Sparkle.framework/Versions/Current/.
Sparkle Test App.app: valid on disk
Sparkle Test App.app: satisfies its Designated Requirement
Mac-1613721063049:Sparkle-1.24.0.tar runner$ 
```

Need to test that codesigning without deleting the translations gives the same issue.

vadi2 committed:
Fetch Sparkle 1.25rc ourselves

vadi2 committed:
Merge branch 'development' into codesign-macos-builds

vadi2 committed:
Ditch Cocoapods

vadi2 committed:
Restore keychain deletion

vadi2 committed:
Cleanup cmakelists

vadi2 committed:
Cleanup build action

vadi2 committed:
Remove debug branch

vadi2 committed:
Up to latest working aqtinstall

vadi2 committed:
Reset cache key

vadi2 committed:
Use installers branch temporarily

vadi2 committed:
Revert "Use installers branch temporarily"

This reverts commit bba711d5a8ee7e6b4e41eb2d1f8940859133bcdc.

vadi2 commented:
Nice, works well:

```
2021-02-20T08:39:50.2475450Z Validating codesigning worked with codesign -vv --deep-verify:
2021-02-20T08:39:51.6628160Z ./Mudlet-4.10.1-testing-pr4746-bba711d5.app: valid on disk
2021-02-20T08:39:51.6632160Z ./Mudlet-4.10.1-testing-pr4746-bba711d5.app: satisfies its Designated Requirement
```

demonnic commented on .github/workflows/build-mudlet.yml:
```diff
@@ -62,7 +62,7 @@ jobs:
       with:
         path: ${{runner.workspace}}/Qt/${{matrix.qt}}
-        key: ${{matrix.os}}-qt-${{matrix.qt}}
+        key: ${{matrix.os}}-qt-${{matrix.qt}}-testtest
```
Don't still want the -testtest do we?

vadi2 replied:
Fixed

vadi2 committed:
Update build-mudlet.yml
